src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1556,3 +1556,13 @@ def progress_callback(acl_changes):
         failed_entries.append(acl_changes.batch_failures)
 
     namespace.progress_hook = progress_callback
+
+
+def get_not_none_validator(attribute_name):
+    def validate_not_none(cmd, namespace):
+        attribute = getattr(namespace, attribute_name, None)
+        options_list = cmd.arguments[attribute_name].type.settings.get('options_list')
```

Comment:
are you sure `options_list` always be there? recommend to give default value as fallback.

Reply:
Yes, all arguments should have `options_list`. Even if no `option_list` specified, core will create `option_list` by `dest` since `dest` is mandatory for each argument. [Code in knack](https://github.com/microsoft/knack/blob/dc7b7d857ce3c11b2a3155b81f2470b3b579e7c9/knack/arguments.py#L45-L68):
```Python
class CLICommandArgument(object):


    NAMED_ARGUMENTS = ['options_list', 'validator', 'completer', 'arg_group', 'deprecate_info', 'preview_info',
                       'experimental_info', 'default_value_source']


    def __init__(self, dest=None, argtype=None, **kwargs):
        """An argument that has a specific destination parameter.

        :param dest: The parameter that this argument is for
        :type dest: str
        :param argtype: The argument type for this command argument
        :type argtype: knack.arguments.CLIArgumentType
        :param kwargs: see knack.arguments.CLIArgumentType
        """
        self.type = CLIArgumentType(overrides=argtype, **kwargs)
        if dest:
            self.type.update(dest=dest)


        # We'll do an early fault detection to find any instances where we have inconsistent
        # set of parameters for argparse
        if not self.options.get('dest', False):
            raise ValueError('Missing dest')
        if not self.options_list:  # pylint: disable=access-member-before-definition
            self.options_list = ('--{}'.format(self.options['dest'].replace('_', '-')),)
```